cdap-proto/src/main/java/io/cdap/cdap/proto/NamespaceMeta.java

```diff
@@ -90,6 +97,14 @@ public static final class Builder {
     private int keytabURIVersion;
     private long generation;
     private Map<String, String> configMap = new HashMap<>();
+    // The repository configuration properties
+    private String repoProvider;
```

Comment:
Why we have all these fields expanded in the builder, but only storing it as a Map inside the namespace repo config class? I think it's better to be consistent in both places. We can use fields instead of a map in the repo config?

Reply:
I think keeping it as properties mapping was because various SCM vendors implementations can be so different, instead of a well defined structure, the configuration should be extensible in future to support other SCM vendors, suggested by @albertshau

Reply:
update: replaced map with commonly defined fields